### PR TITLE
Replace boost vf2 with WSM module in GraphPlacement

### PR DESCRIPTION
**Pull request by sjdilkes**
This commit preserves the current edge breaking routine, but replaces the boost vf2 call with a call to the new weighted subgraph monomorphism module instead.

### CODE REVIEW

File: tket/src/ArchAwareSynth/Path.cpp [hunk 1]
```diff
@@ -165,24 +151,25 @@ std::list<unsigned> PathHandler::find_path(unsigned i, unsigned j) {
...
+  Architecture::UndirectedConnGraph undirected_target =
+      arch.get_undirected_connectivity();
+  QubitGraph::UndirectedConnGraph undirected_pattern =
+      q_graph.get_undirected_connectivity();
+  std::vector<qubit_bimap_t> all_maps = get_unweighted_subgraph_monomorphisms(
+      undirected_pattern, undirected_target, 1, timeout);
 
   /* Architecture has no hampath, sad. */
-  if (!found_monomorphism) return {};
+  if (all_maps.size() == 0) return {};
```
Comment (cqc-alec) [inline tket/src/ArchAwareSynth/Path.cpp:172]:
```suggestion
  if (all_maps.empty()) return {};
```

File: schemas/compiler_pass_v1.json [hunk 2]
```diff
@@ -834,13 +834,13 @@
         "timeout": {
           "type": "integer",
           "minimum": 0,
-          "description": "Timeout for vf2 search, in seconds."
+          "description": "Timeout for monomorphism search, in milliseconds."
```
Comment (cqc-alec) [inline schemas/compiler_pass_v1.json:837]: I assume this was a bug, we aren't changing it?

File: recipes/tket/conanfile.py [hunk 3]
```diff
@@ -106,29 +102,9 @@ def configure(self):
...
-            tools.patch(base_path=boost_include_path, patch_file=patch_file)
         cmake = self._configure_cmake()
-        try:
-            print("Building")
-            cmake.build()
-        finally:
-            for filepath in patches.keys():
-                print("Restoring " + filepath)
-                shutil.move(filepath + ".original", filepath)
+        print("Building")
```
Comment (cqc-alec) [inline recipes/tket/conanfile.py:106]: I'd delete this print line as well as it isn't useful now.

**Review by cqc-alec — CHANGES_REQUESTED** [review]

File: tket/src/ArchAwareSynth/Path.cpp (hunk 1 above)
Comment (sjdilkes) [reply tket/src/ArchAwareSynth/Path.cpp]: Done

File: schemas/compiler_pass_v1.json (hunk 2 above)
Comment (sjdilkes) [reply schemas/compiler_pass_v1.json:837]: Yeah the json was incorrect previously

File: recipes/tket/conanfile.py (hunk 3 above)
Comment (sjdilkes) [reply recipes/tket/conanfile.py:106]: done

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**